olp-cpp-sdk-authentication/src/SignInResultImpl.cpp

```diff
@@ -91,5 +103,63 @@ const std::string& SignInResultImpl::GetScope() const { return scope_; }
 
 bool SignInResultImpl::IsValid() const { return is_valid_; }
 
+std::string SignInResultImpl::ParseJwtToken(const std::string& token) {
+  std::string result;
+  std::string::size_type first_dot = token.find_first_of('.');
+  if (std::string::npos == first_dot) {
+    OLP_SDK_LOG_ERROR_F(kLogTag, "Cannot parse ClientId. Wrong token format.");
+    return result;
```

Comment:
```suggestion
    return std::string{}; // or ""
```
The reason to return empty string is to protect ourself from some mistake when `result` variable might be initialized to some value before by mistake. Also, returning empty string directly improves readability and no need to check what value the `result` can have.

Reply:
I'd also add a line in the public docs that, if there is a parse error, the returned client ID is empty. That will help users to avoid unpleasant surprises with empty client IDs.